src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -272,7 +274,7 @@ public void run() {
 
     @SuppressWarnings("PMD.CollapsibleIfStatements")
     private void initialize() throws IOException {
-        synchronized (this) {
```

Comment:
does this make any difference?

Reply:
Just that synchronizing on public objects is suspect as it breaks encapsulation — so I revised it.